modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -76,111 +91,146 @@ object CurrencySnapshotCreator {
+                    )
+                    .ifM(
+                      service
+                        .combine(lastState, updates.toList)
+                        .flatMap { state =>
+                          service.serializeState(state)
+                        }
+                        .map(_.some)
+                        .handleErrorWith(err =>
+                          logger
+                            .error(err)(
+                              s"Unhandled exception during combine and serialize data application, fallback to last data application"
+                            )
+                            .as(maybeLastDataApplication)
+                        ),
+                      logger.warn("Data application is not valid") >> maybeLastDataApplication.pure[F]
+                    )
+                }
+                .getOrElse(maybeLastDataApplication.pure[F])
+          }.flatSequence
+
+          (acceptanceResult, acceptedRewardTxs, snapshotInfo, stateProof) <- currencySnapshotAcceptanceManager.accept(
+            blocksForAcceptance,
+            lastContext,
+            lastActiveTips,
+            lastDeprecatedTips,
+            transactions =>
+              rewards
+                .map(_.distribute(lastArtifact, lastContext.snapshotInfo.balances, transactions, trigger, events))
+                .getOrElse(SortedSet.empty[RewardTransaction].pure[F])
+          )
+
+          (awaitingBlocks, rejectedBlocks) = acceptanceResult.notAccepted.partitionMap {
+            case (b, _: BlockAwaitReason)     => b.asRight
+            case (b, _: BlockRejectionReason) => b.asLeft
+          }
+
+          (deprecated, remainedActive, accepted) = getUpdatedTips(
+            lastActiveTips,
+            lastDeprecatedTips,
+            acceptanceResult,
+            currentOrdinal
+          )
+
+          (height, subHeight) <- getHeightAndSubHeight(lastArtifact, deprecated, remainedActive, accepted)
+
+          artifact = CurrencyIncrementalSnapshot(
+            currentOrdinal,
+            height,
+            subHeight,
+            lastArtifactHash,
+            accepted,
+            acceptedRewardTxs,
+            SnapshotTips(
+              deprecated = deprecated,
+              remainedActive = remainedActive
+            ),
+            stateProof,
+            currentEpochProgress,
+            maybeNewDataState
+          )
+
+          artifactBinary: Array[Byte] <- artifact.toBinaryF
+          size = artifactBinary.length
+
+          result <-
+            if (size <= maxUnsignedProposalSizeInBytes) {
+              CurrencySnapshotCreationResult(
+                artifact,
+                CurrencySnapshotContext(lastContext.address, snapshotInfo),
+                awaitingBlocks.toSet ++ sizeExcessiveBlocks,
+                rejectedBlocks.toSet
+              ).pure[F]
+            } else {
+              acceptanceResult.accepted match {
+                case _ :+ _ :+ excessivePair =>
```

Comment:
Why not:
```
case acceptedBlocks :+ excessivePair =>
  excessivePair match {
    case (excessiveBlock, _) =>
      val remainingBlocks = acceptedBlocks.map(_._1)
      val excessiveBlocks = sizeExcessiveBlocks + excessiveBlock
      createProposalWithSizeLimit(remainingBlocks, excessiveBlocks)
  }
```
do we consider creating a snapshot without a single event an unacceptable case?

If so we could still pass the currently accepted except the last one, maybe like this:
```
case acceptedBlocks @ (_ :+ _) :+ excessivePair =>
```

I'm not considering it the essential update, just rerunning with possibly ignored blocks we can have in `blocksForAcceptance` seems wasteful.